### PR TITLE
Focus today on opened

### PR DESCRIPTION
**Pull request by peteruithoven**
Selects / focuses in on today when opened. 
Fixes: https://github.com/elementary/wingpanel-indicator-datetime/issues/102

### CODE REVIEW

File: src/Widgets/calendar/Grid.vala [hunk 1]
```diff
@@ -69,6 +69,24 @@ namespace DateTime.Widgets {
             }
         }
 
+        public void set_focus_to_today () {
+            debug ("set_focus_to_today");
+            if (grid_range == null) return;
```
Comment (jeremypw) [inline src/Widgets/calendar/Grid.vala]: Elementary code style:  No single line if clauses.

File: src/Widgets/calendar/Grid.vala [hunk 2]
```diff
@@ -69,6 +69,24 @@ namespace DateTime.Widgets {
...
+            debug ("  found grid_range");
+            Gee.List<GLib.DateTime> dates = grid_range.to_list ();
+            int i = 0;
```
Comment (jeremypw) [inline src/Widgets/calendar/Grid.vala]: `i` can be declared in the for statement as only used inside the loop. 
```
for (int i = 0; i < dates.size; i++) {
```

**Comment by peteruithoven** [comment]
Thanks for the review. Yeah this code could use some serious refactoring, especially data flow wise. But I'm kind of afraid of date related stuff.

**Comment by jeremypw** [comment]
You can lose some lines by using the loop in GridDay:set_range () and checking the days to see if they are 'today' there instead of creating a new loop. You will need to grab_focus for the appropriate day in an Idle loop. You can also avoid the grab_focus_force function by making the 'grab_focus' member into an accessible property.

**Comment by jeremypw** [comment]
There is a question over what should happen when you navigate away from the current month and then back.  Should 'today' be reselected?  I guess not, in which case your solution is better as it only runs on creation of the Calendar View.  Using the set_range () loop would me that 'today' would be selected every time you navigate to the current month.

**Comment by jeremypw** [comment]
I am going to approve this in view of #104 which assumes that events (and the selected day) should not be remembered when navigating between months.

**Comment by peteruithoven** [comment]
> There is a question over what should happen when you  navigate away from the current month and then back.  Should 'today' be  reselected?  I guess not, in which case your solution is better as it  only runs on creation of the Calendar View.  Using the set_range () loop  would me that 'today' would be selected every time you navigate to the  current month.

Yes, it was indeed of my goal to not remember the day when navigating between months.

**Pull request merged**